fn/either.go

```diff
@@ -16,6 +16,17 @@ func NewRight[L any, R any](r R) Either[L, R] {
 	return Either[L, R]{left: None[L](), right: Some(r)}
 }
 
+// ElimEither is the universal Either eliminator. It can be used to safely
+// handle all possible values inside the Either by supplying two continuations,
+// one for each side of the Either.
+func ElimEither[L, R, O any](f func(L) O, g func(R) O, e Either[L, R]) O {
```

Comment:
Q: So we want this to always return the left value first, if it exists ?

Reply:
It is technically a fundamental failure for both of them to be set at once. I was actually going to update the internal representation here to be more CBC than it is currently. So yes if the left value exists it should be the case that the right should not and vice versa. I can add a commit to the end of this that reworks the implementation to enforce that a little bit more structurally.

Reply:
I like the new design.